app/v6/lossClaims/create/model/request/CreateLossClaimRequestData.scala

```diff
@@ -22,5 +22,6 @@ import v6.lossClaims.create.CreateLossClaimSchema
 trait CreateLossClaimRequestData {
   def nino: Nino
   def lossClaim: CreateLossClaimRequestBody
+  def taxYearClaimedFor: String = lossClaim.taxYearClaimedFor
```

Comment:
Less of a request but more of a question for my own understanding. Why create the extra def within the trait to assign the value from the lossClaim rather than using the value straight out of the lossClaim object?

Reply:
One thing to note is that this pattern is also useful when multiple different implementations of `CreateLossClaimRequestBody` might exist, but all must have a `taxYearClaimedFor` field. For example, if a Def 2 that extends it is created with some additional fields. Not likely with losses but it can happen.

Reply:
Also, if for some reason you have to name them differently [here](https://github.com/hmrc/individual-losses-api/blob/MTDSA-28819/app/v6/lossClaims/create/model/request/CreateLossClaimRequestBody.scala#L24) is named `taxYear` i.e. the field name in the case class doesn't match the abstract method in the trait, then you will need to override in the case class like

```
case class Def1_CreateLossClaimRequestBody(taxYearClaimedFor: String, typeOfLoss: TypeOfLoss, typeOfClaim: TypeOfClaim, businessId: String)
    extends CreateLossClaimRequestBody {
  override def taxYear: String = taxYearClaimedFor
}
```